ext/financialacls/financialacls.php

```diff
@@ -185,18 +185,29 @@ function financialacls_civicrm_pre($op, $objectName, $id, &$params) {
  * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_selectWhereClause
  */
 function financialacls_civicrm_selectWhereClause($entity, &$clauses) {
-  if (!financialacls_is_acl_limiting_enabled()) {
-    return;
-  }
-  if ($entity === 'LineItem') {
-    $types = [];
-    CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($types);
-    if ($types) {
+  switch ($entity) {
+    case 'LineItem':
+      if (!financialacls_is_acl_limiting_enabled()) {
+        return;
+      }
+      $types = [];
+      CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($types);
+      if ($types) {
+        $clauses['financial_type_id'] = 'IN (' . implode(',', array_keys($types)) . ')';
+      }
+      else {
+        $clauses['financial_type_id'] = '= 0';
+      }
+      break;
+
+    case 'MembershipType':
+      if (!CRM_Financial_BAO_FinancialType::isACLFinancialTypeStatus()) {
```

Comment:
@mattwire any reason to not use the `financialacls_is_acl_limiting_enabled()` function instead?